lib/translation.ts

```diff
@@ -154,47 +153,29 @@ export function translatePlural(
  * the translations are loaded
  * @return {Promise} promise
  */
-export function loadTranslations(appName: string, callback: (...args: []) => unknown) {
-	interface TranslationBundle {
-		translations: Translations
-		pluralForm: string
-	}
-
-	if (hasAppTranslations(appName) || getLocale() === 'en') {
+export async function loadTranslations(appName: string, callback: (...args: []) => unknown) {
+	if (hasAppTranslations(appName) || getLanguage() === 'en') {
 		return Promise.resolve().then(callback)
 	}
 
-	const url = generateFilePath(appName, 'l10n', getLocale() + '.json')
+	const { generateFilePath } = await import('@nextcloud/router')
+	const url = generateFilePath(appName, 'l10n', getLanguage() + '.json')
 
-	const promise = new Promise<TranslationBundle>((resolve, reject) => {
-		const request = new XMLHttpRequest()
-		request.open('GET', url, true)
-		request.onerror = () => {
-			reject(new Error(request.statusText || 'Network error'))
-		}
-		request.onload = () => {
-			if (request.status >= 200 && request.status < 300) {
-				try {
-					const bundle = JSON.parse(request.responseText)
-					if (typeof bundle.translations === 'object') resolve(bundle)
-				} catch (error) {
-					// error is probably a SyntaxError due to invalid response text, this is handled by next line
-				}
-				reject(new Error('Invalid content of translation bundle'))
-			} else {
-				reject(new Error(request.statusText))
-			}
+	const response = await fetch(url)
+	if (!response.ok) {
+		throw new Error(response.statusText || 'Network error')
+	}
+	try {
+		const bundle = await response.json()
+		if (typeof bundle.translations === 'object') {
+			register(appName, bundle.translations)
+			return Promise.resolve(bundle).then(callback)
 		}
-		request.send()
-	})
-
-	// load JSON translation bundle per AJAX
-	return promise
-		.then((result) => {
-			register(appName, result.translations)
-			return result
-		})
-		.then(callback)
+		throw new Error('Invalid content of translation bundle')
```

Comment:
I personally prefer error handling instead of success handling

```ts
		if (typeof bundle.translations !== 'object') {
			throw new Error('Invalid content of translation bundle')
		}

		register(appName, bundle.translations)
		return callback(bundle)